src/items/unions.md

```diff
@@ -37,13 +39,17 @@ struct fields:
 let f = u.f1;
 ```
 
+## Reading and writing union fields
+
 Unions have no notion of an "active field". Instead, every union access just
 interprets the storage at the type of the field used for the access. Reading a
-union field reads the bits of the union at the field's type. It is the
-programmer's responsibility to make sure that the data is valid at that
-type. Failing to do so results in undefined behavior. For example, reading the
-value `3` at type `bool` is undefined behavior. Effectively, writing to and then
-reading from a union is analogous to a [`transmute`] from the type used for
+union field reads the bits of the union at the field's type. Fields might have a
+non-zero offset (except when `#[repr(C)]` is used); in that case the bits
+starting at the offset of the fields are read. It is the programmer's
+responsibility to make sure that the data is valid at the field's type. Failing
```

Comment:
It's not really the role of the reference to say "it is the programmer's responsibility" or "this is how you use X" but it's fine here imo.

Reply:
Do you have a suggestion for wording it differently? I think it is useful to think in terms of responsibilities/obligations when talking about unsafe code, and hence I think it is useful for the reference to explain it as such.

I also have 0 experience writing standard-esque text.^^ Given my experience *reading* it though, it seems often so useless that I am not sure I am sad about that...

Reply:
Not really; the text itself is good but it's probably more aimed at users (e.g. the nomicon) whereas the reference is intended as the seeds of a spec with just definitions of the language but not the "why" and "this is how I use it". That is, should be more like the details of one of your academic papers with operational semantics and judgements... it isn't now, but it should be. :)

Reply:
You mean the details that only 2 people on this planet read because they are incomprehensible? ;)

Reply:
That's not the papers I read :)

I think something like the WASM spec has a nice structure:
- https://webassembly.github.io/spec/core/index.html